### PR TITLE
Update sort command to allow sorting by most fields

### LINKED ISSUE #91: Generalise use of SortCommand to more fields

### PR DESCRIPTION
**Pull request by ylyma**
Closes #91 

Allow sorting by name, phone, email, length of comment, number of tags and GPA.
Allow SortCommand to be extended easily to accommodate more fields when they are added.

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Address.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * Returns true if a given string is a valid email.
+     */
+    public static boolean isValidAddress(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    @Override
+    public String toString() {
+        return value;
```
Comment (ravern) [inline src/main/java/seedu/address/model/person/Address.java:41]: Can I check why was `Address.java` added in this PR?

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ * Sorts all applicants by the given field name.
+ */
+public class SortCommand extends Command {
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sorts all applicants by the given field name.\n"
+            + "Parameters: FIELD_NAME\n"
+            + "Example: " + COMMAND_WORD + " -gpa";
```
Comment (ravern) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:21]: Is the `-` in `-gpa` a typo?

**Review by ravern — CHANGES_REQUESTED** [review]
Requesting changes because of `Address.java`

File: src/main/java/seedu/address/model/person/Address.java (hunk 1 above)
Comment (ylyma) [reply src/main/java/seedu/address/model/person/Address.java:41]: My bad, I think I pulled wrongly.

**Review by ravern — APPROVED** [review]
Swee Chai Butterfly 🦋🦋

**Pull request merged**